File: php/1403.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function minSubsequence($nums) {
        $ret = [];
        rsort($nums);

        $middle = (int)(array_sum($nums) / 2);

        $sum = 0;
        for ($i = 0; $i < count($nums); $i++) {
            $ret[] = $nums[$i];
            $sum += $nums[$i];
            if ($sum > $middle) {
                break;
            }
        }

        return $ret;
    }
}

$solution = new Solution();
var_dump($solution->minSubsequence([4, 4, 7, 6, 7]));

File: go/greedy/1403.go
package main

import "sort"

func minSubsequence(nums []int) []int {
	sort.Sort(sort.Reverse(sort.IntSlice(nums)))

	sum := 0
	for _, v := range nums {
		sum += v
	}
	middle,index, cur := sum / 2,0, 0
	for {
		if index == len(nums) {
			break
		}
		cur += nums[index]
		if cur > middle {
			break
		}
		index++
	}
	return nums[:index+1]
}
